File: backend/src/main/java/com/bloghub/api/service/AuthService.java
package com.bloghub.api.service;

import com.bloghub.api.dto.JwtResponse;
import com.bloghub.api.dto.LoginRequest;
import com.bloghub.api.dto.LogoutRequest;
import com.bloghub.api.dto.RefreshTokenRequest;
import com.bloghub.api.dto.RegisterRequest;
import com.bloghub.api.dto.UserDto;

/**
 * Contract for authentication operations.
 */
public interface AuthService {

    /** Register a new user and return their profile */
    UserDto register(RegisterRequest request);

    /** Authenticate credentials and return a JWT token */
    JwtResponse login(LoginRequest request);

    /** Authenticate credentials and return an access + refresh token pair */
    JwtResponse login(LoginRequest request, String clientIp);

    /** Rotate a refresh token and issue a new access token */
    JwtResponse refreshToken(RefreshTokenRequest request, String clientIp);

    /** Revoke the current access token and optional refresh token */
    void logout(String authorizationHeader, LogoutRequest request);
}

File: frontend/src/services/authService.js
import apiClient from './apiClient'

const authService = {
  /** Register a new user */
  register: async (data) => {
    const response = await apiClient.post('/auth/register', data)
    return response.data
  },

  /** Login and receive JWT token */
  login: async (credentials) => {
    const response = await apiClient.post('/auth/login', credentials)
    return response.data
  },

  /** Get current user's profile */
  getCurrentUser: async () => {
    const response = await apiClient.get('/users/me')
    return response.data
  },

  /** Update user profile */
  updateProfile: async (userId, data) => {
    const response = await apiClient.put(`/users/${userId}`, data)
    return response.data
  },
}

export default authService
